src/ifcopenshell-python/ifcopenshell/util/constants.py

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+# ----------------------------------------------------------------
+# IMPORTS
+# ----------------------------------------------------------------
+
+from typing import Literal
+
+# ----------------------------------------------------------------
+# EXPORTS
+# ----------------------------------------------------------------
+
+__all__ = [
+    "IFC_APPLICATION",
+    "IFC_GRID_AXIS",
+    "IFC_PRODUCT",
+    "IFC_PROJECT",
+    "IFC_PROPERTY_SINGLE_VALUE",
+    "IFC_QUANTITY_AREA",
+    "IFC_QUANTITY_COUNT",
+    "IFC_QUANTITY_LENGTH",
+    "IFC_QUANTITY_TIME",
+    "IFC_QUANTITY_VOLUME",
+    "IFC_QUANTITY_WEIGHT",
+    "IFC_SI_UNIT",
+    "IFC_TYPES",
+    "IFC_UNIT_ASSIGNMENT",
+]
+
+# ----------------------------------------------------------------
+# CONSTANTS
+# ----------------------------------------------------------------
+
+# TODO: complete this
+IFC_APPLICATION = "IfcApplication"
+IFC_GRID_AXIS = "IfcGridAxis"
+IFC_PRODUCT = "IfcProduct"
+IFC_PROJECT = "IfcProject"
+IFC_PROPERTY_SINGLE_VALUE = "IfcPropertySingleValue"
+IFC_QUANTITY_AREA = "IfcQuantityArea"
+IFC_QUANTITY_COUNT = "IfcQuantityCount"
+IFC_QUANTITY_LENGTH = "IfcQuantityLength"
+IFC_QUANTITY_TIME = "IfcQuantityTime"
+IFC_QUANTITY_VOLUME = "IfcQuantityVolume"
+IFC_QUANTITY_WEIGHT = "IfcQuantityWeight"
+IFC_SI_UNIT = "IfcSIUnit"
+IFC_UNIT_ASSIGNMENT = "IfcUnitAssignment"
+
+# TODO: complete this
+IFC_TYPES = Literal[
+    "IfcApplication",
+    "IfcGridAxis",
+    "IfcProduct",
+    "IfcProject",
+    "IfcPropertySingleValue",
+    "IfcQuantityArea",
+    "IfcQuantityCount",
+    "IfcQuantityLength",
+    "IfcQuantityTime",
+    "IfcQuantityVolume",
+    "IfcQuantityWeight",
+    "IfcSIUnit",
+    "IfcUnitAssignment",
+]
```

Comment:
What are all these constants for?

Reply:
The intention there was to create a type. I should add that in my PR this is not _directly_ used.

Usage of this type would allow any developer who imports of `ifcopenshell` to for example write methods like this:
```py
from functools import reduce
from ifcopenshell.file import file as IfcFile
from ifcopenshell.util.constants import IFC_TYPES

def get_names_of_entities_by_types(
  f: IfcFile,
  types: set[IFC_TYPES],
  /,
) -> list[str]:
  """
  Obtains all entities by a given set of types
  and returns the names of these elements
  """
  # get all entities
  entities = reduce(
    lambda x, y: x + y,
    map(lambda t: f.by_type(t), types),
    []
  )
  # get names
  names = map(lambda e: getattr(e, "Name", None), entities)
  # filter out null-values
  names = filter(lambda name: isinstance(name, str), names)
  # return list
  return list(names)
```
This provides the advantage to the developer as one then has type hints, so that one nows what values can be used:
![example](https://github.com/user-attachments/assets/82482c9d-9432-4c9e-9683-865ac2e28fba)
In fact, it would by nice at some point to implement such a `Literal`-type for methods in the repository like `(class file).by_type`, etc.